### ISSUE TITLE
[Question] Can remove `dateutil` from dependency?

### ISSUE THREAD

**Issue by isac322** [arrow-py/arrow#1094]
## Feature Request

Python 3.9 introduced `ZoneInfo` and it backported 3.6+ (https://pypi.org/project/backports.zoneinfo/).
Looking at the code, except for `dateutil.rrule` and `dateutil.relativedelta`, it seems that `dateutil` is being used for some reason related to timezone. Can `ZoneInfo` replace the part and delete `dateutil`?

**Comment by anishnya** [comment]
@jadchaar, @krisfremen and @systemcatch are more well-versed on this than I am. From what I recall, there was talk about removing dateutil as dependency earlier last year.

**Comment by jadchaar** [comment]
Yup, it would be great to lose a dependency wit ZoneInfo, but we'd need to port over rrule and relativedelta or find potential alternatives before dropping the library. A PR would be appreciate @isac322 :)!

**Comment by jadchaar** [comment]
Related to:
https://github.com/arrow-py/arrow/issues/825
https://github.com/arrow-py/arrow/issues/825
